### PR TITLE
no need for default orig_size stored for invertible transforms

### PR DESCRIPTION
**Pull request by rijobro**
### Description
If an object didn't have the attribute `shape`, we stored `orig_size` as `None`. However, `None` can't be collated into a batch, so if the object doesn't have the attribute `shape`, then don't store that information.

Fixes https://github.com/Project-MONAI/MONAI/discussions/2414.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.

### CODE REVIEW

File: monai/transforms/inverse.py [hunk 1]
```diff
@@ -76,8 +76,11 @@ def push_transform(
         info = {
             InverseKeys.CLASS_NAME: self.__class__.__name__,
             InverseKeys.ID: id(self),
-            InverseKeys.ORIG_SIZE: orig_size or (data[key].shape[1:] if hasattr(data[key], "shape") else None),
         }
+        if orig_size is not None:
+            InverseKeys.ORIG_SIZE = orig_size
+        elif hasattr(data[key], "shape"):
+            InverseKeys.ORIG_SIZE = data[key].shape[1:]
```
Comment (wyli) [inline monai/transforms/inverse.py:83]: missing the `else` here for the -1?

Comment (rijobro) [reply monai/transforms/inverse.py:83]: sorry, i updated the docs. There's no reason to have a default (None or -1). If the object doesn't have `shape` then none of that batch will either. So no need to add anything. If shape doesn't exist in the forward direction, the orig_size won't be used in the inverse direction, either.

**Review by wyli — APPROVED** [review]
thanks!

File: monai/transforms/inverse.py (hunk 1 above)
Comment (rijobro) [reply monai/transforms/inverse.py:83]: ...or that's what I thought until the CI/CD failed...

**Pull request merged**